cmd/kots/cli/install.go

```diff
@@ -162,6 +162,28 @@ func InstallCmd() *cobra.Command {
 			}()
 
 			upstream := pull.RewriteUpstream(args[0])
+			preferredChannelSlug, err := extractPreferredChannelSlug(upstream)
+			if err != nil {
+				return errors.Wrap(err, "failed to extract preferred channel slug")
+			}
+
+			// fetch the latest version of the license to verify channel access if online install
+			if license != nil {
```

Comment:
the install command is already long and complex, what do you think about moving this logic be to a separate package?